src/Rules/Operators/InvalidBinaryOperationRule.php

```diff
@@ -124,4 +157,16 @@ public function processNode(Node $node, Scope $scope): array
 		];
 	}
 
+	private function isDivisionByZeroCaught(Node $node): bool
```

Comment:
inspired by https://github.com/phpstan/phpstan-src/blob/64d4a3b61c81aced05fa690bb7a418d34bfb4213/src/Rules/Comparison/MatchExpressionRule.php#L168-L181

Reply:
Nice.

The only remaining question is to decide if
```
/** @throws DivisionByZeroError */
function foo() {
        $i / $x;
}
```
should be reported or not.

Reply:
I have this already in the pipeline :).

it might make sense to extract the "isXXExceptionCaught" and "hasXXExceptionThrowsTag" into a service to ease re-use